colrev/ops/built_in/load_conversion/ris_rispy_loader.py

```diff
@@ -32,82 +33,101 @@ class RisRispyLoader(JsonSchemaMixin):
+        "number",
+    }
+    replace_with = {
+        "primary_title": "title",
+        "notes_abstract": "abstract",
+        "secondary_title": "journal",
+        "publication_year": "year",
+    }
+
     def __init__(
         self,
         *,
         load_operation: colrev.ops.load.Load,
         settings: dict,
     ) -> None:
+        self.entries = None
         self.settings = self.settings_class.load_settings(data=settings)
 
     def load(
         self, load_operation: colrev.ops.load.Load, source: colrev.settings.SearchSource
     ) -> dict:
         """Load records from the source"""
-        records = {}
+        records: dict = {}
         if source.filename.is_file():
-            records = self.__parse_ris_file_and_convert_to_bib(source)
+            with open(source.filename, encoding="utf-8") as ris_file:
+                self.entries = rispy.load(file=ris_file)
+            if self.entries:
+                records = self.__parse_ris_file_and_convert_to_bib()
+
         endpoint_dict = load_operation.package_manager.load_packages(
             package_type=colrev.env.package_manager.PackageEndpointType.search_source,
             selected_packages=[source.get_dict()],
             operation=load_operation,
             ignore_not_available=False,
         )
         endpoint = endpoint_dict[source.endpoint]
-
         records = endpoint.load_fixes(  # type: ignore
             load_operation, source=source, records=records
         )
         return records
 
-    def __parse_ris_file_and_convert_to_bib(
-        self, source: colrev.settings.SearchSource
-    ) -> dict:
-        """Parses given ris file with rispy and convert it to bib record"""
-
-        def add_entry(_entry: dict, key: str, _record: dict) -> None:
-            if key in _entry:
-                _record[key] = _entry[key]
+    def __get_entry_id(self, entry: dict) -> typing.Tuple[str, str, int]:
+        try:
+            authors = entry["first_authors"] + entry["secondary_authors"]
+        except KeyError:
+            authors = entry["authors"]
+        first_three_authors = authors
+        if len(authors) >= 3:
+            first_three_authors = authors[:3]
+        try:
+            year = entry["year"].strip("/")
+        except KeyError:
+            year = entry["publication_year"].strip("/")
+        three_author = "".join(
+            [author.rsplit(",")[0].strip() for author in first_three_authors]
+        )
+        _id = f"{three_author.lower()}{year}"
+        return _id, authors, year
 
-        with open(source.filename, encoding="utf-8") as ris_file:
-            entries = rispy.load(file=ris_file)
+    def __add_entry(self, _entry: dict, key: str, record: dict) -> None:
+        if key in _entry:
+            val = _entry[key]
+            target_key = key if key not in self.replace_with else self.replace_with[key]
+            record[target_key] = val
 
+    def __parse_ris_file_and_convert_to_bib(self) -> dict | None:
```

Comment:
Maybe just `__parse_ris_file()`? The records are saved to a bib file afterwards.

Reply:
Okay